blueprints/data-solutions/data-platform-foundations/03-orchestration.tf

```
@@ -91,9 +91,13 @@ module "orch-project" {
     "storage.googleapis.com",
     "storage-component.googleapis.com"
   ])
-  service_encryption_key_ids = {
-    composer = [try(local.service_encryption_keys.composer, null)]
-    storage  = [try(local.service_encryption_keys.storage, null)]
+  service_agent_encryption_key_ids = {
+    composer         = compact([var.service_encryption_keys.composer])
+    artifactregistry = compact([var.service_encryption_keys.composer])
+    container-engine = compact([var.service_encryption_keys.composer])
+    compute          = compact([var.service_encryption_keys.composer])
+    pubsub           = compact([var.service_encryption_keys.composer])
+    storage          = compact([var.service_encryption_keys.composer, var.service_encryption_keys.storage])
```

```suggestion
    artifactregistry = compact([var.service_encryption_keys.artifactregistry])
    container-engine = compact([var.service_encryption_keys.container-engine])
    compute          = compact([var.service_encryption_keys.compute])
    pubsub           = compact([var.service_encryption_keys.pubsub])
    storage          = compact([var.service_encryption_keys.storage, var.service_encryption_keys.storage])
```
? Or otherwise it needs a comment, why we are passing the same key here.

Or if we are at it, maybe to avoid such bugs:
```suggestion
    for sa in ["composer", "artifactregistry", "container-engine", "compute", "pubsub", "storage"]: sa => compact([lookup(var.service_encryption_keys, sa, null)])
    ```

---

Oh, I see the variables now, understood.

---

I got lost, are we good?